### DISCUSSION TITLE
LinkAce installation problems on Apache server (without Docker)

### DISCUSSION THREAD

**Discussion by AlbertoV3** [Kovah/LinkAce#670] [Support]
Hello,
I'm trying install LinkAce on my Apache web server (php8 is installed) without Docker. 
But nothing happens on my screen :(

This is the step where I failed the configuration : 

    Run the following command to generate a secret key for your application and prepare LinkAce for the setup:

    php artisan key:generate

Well, when I try this, nothing happens (the line 5 of .env file is still "APP_KEY=someRandomStringWith32Characters" :/

Do you have any idea what I can fix it ?

I opens a [question in Stackoverflow](https://stackoverflow.com/questions/76749053/linkace-installation-problems-on-server-without-docker) I'm not alone in this case.

Thank you so much in advance,

**Comment by Kovah** [comment]
Hi,
What exactly means "Well, when I try this, nothing happens"? There should either be some success message or an error. Please also do a `ls -la .env` and post the output.

**Reply by AlbertoV3** [reply]
Hi,
Thank you for your answer, this the following output : `-rw-r--r-- 1`

Regards,

**Reply by Kovah** [reply]
Is there some output missing? Because it should look more like this: 
`-rw-r--r--@ 1 user  group  4794 Jul 13 08:34 .env`
Also, please explain what the "nothing happens" means.

**Reply by AlbertoV3** [reply]
Thank you Kovah for your answer.
"Nothing happens" because after running "php artisan key:generate" the .env file was unchanged and also http://myurl.com/public = blank page.

Sorry, this is the following output :
`-rw-r--r-- 1 alberto alberto 727 23 julio. 17:08 .env`

**Reply by Kovah** [reply]
Yeah I understand, but what it the output of the `php artisan key:generate` command? Is there any output? If something happened, it should return something like ` INFO  Application key set successfully.` to your console. If that is not what you get there might be another issue.

`http://myurl.com/public` would also be the wrong directory, the web server should point to the `public` directory as described in the documentation, not the parent directory where LinkAce was installed.

Please post more details about your server where you want to install LinkAce, otherwise this is going to take forever to replicate.
- Operating system with exact version and kernel if applicable
- exact Apache version, the vHost config for LinkAce and all loaded modules (`apache2ctl -M`)
- exact PHP version and all loaded modules (`php -m`)
- directory where you put LinkAce into, there do a `ls -la` and a `ls -la storage`

**Reply by AlbertoV3** [reply]
Thanks for your answers, here are mines : 
"Yeah I understand, but what it the output of the php artisan key:generate command? Is there any output?" => I don't have any output :
[alberto@cervelle myurl.com]$ php artisan key:generate
[alberto@cervelle myurl.com]$

"http://myurl.com/public would also be the wrong directory, the web server should point to the public directory as described in the documentation, not the parent directory where LinkAce was installed."
=> Well on that point, I'm not sure I understand.
What does it mean "web server should point to the public directory" (I read the doc, but it's possible the mistake is here in my case).

"Operating system with exact version and kernel if applicable" => I can't find this information
"exact Apache version, the vHost config for LinkAce and all loaded modules (apache2ctl -M)" => Same here, `apache2ctl -M` : 
bash: apache2ctl: command not found

PHP Version => 8.1.21

[PHP Modules]
bcmath
bz2
calendar
Core
ctype
curl
date
dom
exif
fileinfofilter
ftp
gd
gettext
hash
iconv
imagick
imap
intl
json
ldap
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
pdo_pgsql
pdo_sqlite
pgsql
Phar
posix
readline
Reflection
session
shmop
SimpleXML
soap
sockets
SPL
sqlite3
standard
tidy
tokenizer
xml
xmlreader
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache

Result of `ls -la` :
total 1000
drwxr-x--- 13 alberto nobody   4096 23 juli. 17:08 .
drwx--x--x 73 alberto alberto    4096 23 juli. 17:41 ..
drwxr-xr-x 14 alberto alberto    4096 23 juli. 17:07 app
-rw-r--r--  1 alberto alberto    1686 23 juli. 17:06 artisan
drwxr-xr-x  3 alberto alberto    4096 23 juli. 17:07 bootstrap
-rw-r--r--  1 alberto alberto      92 23 juli. 17:06 .codeclimate.yml
-rw-r--r--  1 alberto alberto    2530 23 juli. 17:06 composer.json
-rw-r--r--  1 alberto alberto  476694 23 juli. 17:06 composer.lock
drwxr-xr-x  2 alberto alberto    4096 23 juli. 17:07 config
-rw-r--r--  1 alberto alberto    2905 23 juli. 17:06 CONTRIBUTING.md
-rw-r--r--  1 alberto alberto     126 23 juli. 17:06 crowdin.yml
drwxr-xr-x  5 alberto alberto    4096 23 juli. 17:07 database
-rw-r--r--  1 alberto alberto     960 23 juli. 17:06 docker-compose.production-simple.yml
-rw-r--r--  1 alberto alberto    1352 23 juli. 17:06 docker-compose.production.yml
-rw-r--r--  1 alberto alberto    1418 23 juli. 17:06 docker-compose.yml
-rw-r--r--  1 alberto alberto     225 23 juli. 17:06 .dockerignore
-rw-r--r--  1 alberto alberto     213 23 juli. 17:06 .editorconfig
-rw-r--r--  1 alberto alberto     727 23 juli. 17:08 .env
-rw-r--r--  1 alberto alberto    1483 23 juli. 17:06 .env.docker
-rw-r--r--  1 alberto alberto    1061 23 juli. 17:06 .env.docker.production
-rw-r--r--  1 alberto alberto     727 23 juli. 17:06 .env.example
-rw-r--r--  1 alberto alberto     111 23 juli. 17:06 .gitattributes
drwxr-xr-x  4 alberto alberto    4096 23 juli. 17:06 .github
-rw-r--r--  1 alberto alberto     356 23 juli. 17:06 .gitignore
drwxr-xr-x 14 alberto alberto    4096 23 juli. 17:07 lang
-rw-r--r--  1 alberto alberto   35149 23 juli. 17:06 LICENSE.md
-rw-r--r--  1 alberto alberto    1566 23 juli. 17:06 nginx.conf
-rw-r--r--  1 alberto alberto    1718 23 juli. 17:06 nginx-ssl.conf
-rw-r--r--  1 alberto alberto     946 23 juli. 17:06 package.json
-rw-r--r--  1 alberto alberto  345946 23 juli. 17:06 package-lock.json
-rw-r--r--  1 alberto alberto     631 23 juli. 17:06 phpcs.xml
-rw-r--r--  1 alberto alberto    1644 23 juli. 17:06 phpunit.xml
drwxr-xr-x  3 alberto alberto    4096 23 juli. 17:07 public
-rw-r--r--  1 alberto alberto    5631 23 juli. 17:06 README.md
drwxr-xr-x  5 alberto alberto    4096 23 juli. 17:07 resources
drwxr-xr-x  2 alberto alberto    4096 23 juli. 17:08 routes
-rw-r--r--  1 alberto alberto    2253 23 juli. 17:06 SECURITY.md
drwxr-xr-x  6 alberto alberto    4096 23 juli. 17:08 storage
drwxr-xr-x  8 alberto alberto    4096 23 juli. 17:08 tests
-rw-r--r--  1 alberto alberto    2368 23 juli. 17:06 update-docker.sh
-rw-r--r--  1 alberto alberto     467 23 juli. 17:06 webpack.mix.js

Result of `ls -la storage` :
total 24
drwxr-xr-x  6 alberto alberto  4096 23 juli. 17:08 .
drwxr-x--- 13 simon nobody 4096 23 juli. 17:08 ..
drwxr-xr-x  5 alberto alberto  4096 23 juli. 17:08 app
drwxr-xr-x  2 alberto alberto  4096 23 juli. 17:08 debugbar
drwxr-xr-x  6 alberto alberto  4096 23 juli. 17:08 framework
drwxr-xr-x  2 alberto alberto  4096 23 juli. 17:08 logs

**Reply by martadinata666** [reply]
How do you get the files? I saw missing files compared to the release extract. Is this `git clone`?

**Reply by AlbertoV3** [reply]
Oh yes ? 
Nope, I directly downloaded the following zip file https://github.com/Kovah/LinkAce/releases/download/v1.12.2/linkace-v1.12.2.zip and upload it into the server.

**Reply by Kovah** [reply]
Alright...

There is no output of the command, so I assume there is some kind of error happening and failing silently. Please add `APP_DEBUG=true` to the .env file and try again, hope this helps in finding the actual error.

If you can't find any details about the operating system, please post _any_ information about where you run LinkAce. I can't help you any further without any details.
You linked a Stackoverflow issue where cPanel is mentioned. If this is the case for you, I am sorry but won't support any of these web hosting panels. They are an absolute pain to work with.

PS: it is correct that the download package misses some files that are in the repository, because they are not needed for the application to run.

**Reply by martadinata666** [reply]
then for some reason, you are missing something on copy, as your `vendor` folder is missing. That is why I'm asking if you are doing `git clone` as it is essentially incomplete.
```
❯ ls -al;
total 936
drwxr-xr-x 1 dedyms dedyms    862 Jul 28 15:00 ./
drwxr-xr-x 1 dedyms dedyms     34 Jul 28 15:00 ../
-rw-r--r-- 1 dedyms dedyms     92 Jul 20 14:43 .codeclimate.yml
-rw-r--r-- 1 dedyms dedyms    225 Jul 20 14:43 .dockerignore
-rw-r--r-- 1 dedyms dedyms    213 Jul 20 14:43 .editorconfig
-rw-r--r-- 1 dedyms dedyms   1483 Jul 20 14:43 .env.docker
-rw-r--r-- 1 dedyms dedyms   1061 Jul 20 14:43 .env.docker.production
-rw-r--r-- 1 dedyms dedyms    727 Jul 20 14:43 .env.example
-rw-r--r-- 1 dedyms dedyms   2905 Jul 20 14:43 CONTRIBUTING.md
-rw-r--r-- 1 dedyms dedyms  35149 Jul 20 14:43 LICENSE.md
-rw-r--r-- 1 dedyms dedyms   5631 Jul 20 14:43 README.md
-rw-r--r-- 1 dedyms dedyms   2253 Jul 20 14:43 SECURITY.md
drwxr-xr-x 1 dedyms dedyms    176 Jul 20 14:43 app/
-rwxr-xr-x 1 dedyms dedyms   1686 Jul 20 14:43 artisan*
drwxr-xr-x 1 dedyms dedyms     24 Jul 20 14:43 bootstrap/
-rw-r--r-- 1 dedyms dedyms   2530 Jul 20 14:43 composer.json
-rw-r--r-- 1 dedyms dedyms 476694 Jul 20 14:43 composer.lock
drwxr-xr-x 1 dedyms dedyms    506 Jul 20 14:43 config/
-rw-r--r-- 1 dedyms dedyms    126 Jul 20 14:43 crowdin.yml
drwxr-xr-x 1 dedyms dedyms     52 Jul 20 14:43 database/
-rw-r--r-- 1 dedyms dedyms   1352 Jul 20 14:43 docker-compose.production.yml
-rw-r--r-- 1 dedyms dedyms   1418 Jul 20 14:43 docker-compose.yml
-rw-r--r-- 1 dedyms dedyms    960 Jul 20 14:43 docker-compose.production-simple.yml
drwxr-xr-x 1 dedyms dedyms    132 Jul 20 14:43 lang/
-rw-r--r-- 1 dedyms dedyms   1566 Jul 20 14:43 nginx.conf
-rw-r--r-- 1 dedyms dedyms   1718 Jul 20 14:43 nginx-ssl.conf
-rwxr-xr-x 1 dedyms dedyms    946 Jul 20 14:43 package.json*
-rw-r--r-- 1 dedyms dedyms 345946 Jul 20 14:43 package-lock.json
-rw-r--r-- 1 dedyms dedyms    631 Jul 20 14:43 phpcs.xml
-rw-r--r-- 1 dedyms dedyms   1644 Jul 20 14:43 phpunit.xml
drwxr-xr-x 1 dedyms dedyms    502 Jul 20 14:43 public/
drwxr-xr-x 1 dedyms dedyms     34 Jul 20 14:43 resources/
drwxr-xr-x 1 dedyms dedyms     52 Jul 20 14:43 routes/
drwxr-xr-x 1 dedyms dedyms     48 Jul 20 14:43 storage/
drwxr-xr-x 1 dedyms dedyms    190 Jul 20 14:43 tests/
-rwxr-xr-x 1 dedyms dedyms   2368 Jul 20 14:43 update-docker.sh*
drwxr-xr-x 1 dedyms dedyms    702 Jul 20 14:43 vendor/
-rw-r--r-- 1 dedyms dedyms    467 Jul 20 14:43 webpack.mix.js
```

**Reply by Kovah** [reply]
Nice catch @martadinata666!
I have downloaded the latest release package from https://github.com/Kovah/LinkAce/releases/download/v1.12.2/linkace-v1.12.2.zip
and the vendor folder is there. @AlbertoV3 can you check your downloaded package?

**Reply by AlbertoV3** [reply]
Hi,

Thanks for your return, I downloaded again the latest package https://github.com/Kovah/LinkAce/releases/download/v1.12.2/linkace-v1.12.2.zip
Now I have the vendor folder.

I'll delete all the files, download the last package and keep you posted ;)

**Reply by AlbertoV3** [reply]
Hi again,

Ok so now I uploaded all files and I this is the output of `php artisan key:generate` command :
php artisan key:generate


                    APPLICATION IN PRODUCTION.


  Do you really wish to run this command? (yes/no) [no]
❯ y

   INFO  Application key set successfully.

So now what I understand is to go on myurl.com/public/ (I've a redirection to public/setup/start) and follow instructions (all requirements are OK, I supposed have to configure database settings) ?

**Reply by Kovah** [reply]
Looks good, but please configure Apache to point to the LinkAce directory, not the public directory inside it.
It depends on how you configure your Apache, so please consult the documentation. If you edit the config directly, it should look like this:
```
<VirtualHost *:80>
    DocumentRoot "/var/www/linkace"
    ServerName www.example.com

    # Other directives here
</VirtualHost>
```
See https://httpd.apache.org/docs/2.4/vhosts/examples.html for details.

**Reply by AlbertoV3** [reply]
Ok, I filled that into the .htaccess file (/public/.htaccess) :
```
#configure Apache to point to the LinkAce directory
<VirtualHost *:80>
    DocumentRoot "/home/alberto/myurl.com"
    ServerName myurl.com

    # Other directives here
</VirtualHost>

<IfModule mod_rewrite.c>
    <IfModule mod_negotiation.c>
        Options -MultiViews -Indexes
    </IfModule>

    RewriteEngine On

    # Handle Authorization Header
    RewriteCond %{HTTP:Authorization} .
    RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]

    # Redirect Trailing Slashes If Not A Folder...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_URI} (.+)/$
    RewriteRule ^ %1 [L,R=301]

    # Handle Front Controller...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_FILENAME} !-f
    RewriteRule ^ index.php [L]
</IfModule>

```
Now, when I check https://myurl.com/, the folder named "public" is "hidden" (before editing the .htaccess file, I saw it)
But https://myurl.com/public/ repond 500 httpcode

Here's my latest log server message: 
[Fri Jul 28 13:07:56.978603 2023] [core:alert] [pid 25001:tid 22465947907840] [client xx.xx.xx.x:0] /home/alberto/myurl.com/public/.htaccess: <VirtualHost not allowed here

**Reply by Kovah** [reply]
Please consult the documentation of your hosting or contact your hosting provider for help. The configuration must go into the virtual host config, not into the htaccess file.

**Reply by AlbertoV3** [reply]
Ok, I removed the lines of .htaccess file and I tried to skip this step (I can't access into the virtual host config).

I managed to following steps : configure database, create account.
But at the end, when I click on "create account", I've the following message : 
"LinkAce
500 - Server Error
An internal server error occured. If you are the administrator, consult the application logs for details."
(I'm on myurl.com/public/setup/account)

**Comment by AlbertoV3** [comment]
Hello again,

News update about my configuration to help other persons in my case.
Spoiler : I've right done the installation and configuration part1 🥳 

I all delete and install again (and again), this is all what I did since my last message : 
- Update PHP version to 8.2 (instead of 8)
- Point to the public folder (on my server it was via Cpanel > Domains)
- Try to `git clone` the repo https://github.com/Kovah/LinkAce.git : the vendor folder is still missing
- So I have downloaded the latest release package from https://github.com/Kovah/LinkAce/releases/download/v1.12.2/linkace-v1.12.2.zip
- And launch the following steps : 
  - Make a copy of the .env.example file and name it .env. 
  - php artisan key:generate
  - ..

Regards,
